### DISCUSSION TITLE
BT-157-LotsGroup subtypes missing currency selector definition

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#542] [Q&A] [answered]
any notice-type where is this field used for example 16
its possible to find under "id" : "GR-LotsGroup-FrameworkAgreement",

missing field definition for currency selection. which is required during validation process 
view generated based on SDK 1.7 (so its possible to see that option where user can pick currency is missing)
![image](https://github.com/OP-TED/eForms-SDK/assets/11737559/d8c8202b-3338-4870-b2d8-41c666caf87a)

result xml from this form
![image](https://github.com/OP-TED/eForms-SDK/assets/11737559/ef92fba4-9fa9-4cdf-a5a2-871ac7919985)

expected XML 
![image](https://github.com/OP-TED/eForms-SDK/assets/11737559/f4b6ddb8-0496-4c5e-af69-813010fc2304)

error message from cvs server, which unhide this missing field in any-notice-subtype.json
![image](https://github.com/OP-TED/eForms-SDK/assets/11737559/e23c5fa7-4532-4ec8-a5f9-5a5b9dbca3fe)

**Comment by rouschr** [comment, marked as answer]
Hello,
you are not the first one to wonder about this:
https://github.com/OP-TED/eForms-SDK/discussions/483
https://github.com/OP-TED/eForms-SDK/discussions/421
https://github.com/OP-TED/eForms-SDK/discussions/121
https://github.com/OP-TED/eForms-SDK/discussions/139
One thing you can do is see how enotices2 is dealing with this.
Historically this was implemented by enotices2 as "implicit", we do have bits in the documentation of the SDK here:

https://docs.ted.europa.eu/eforms/latest/fields/index.html#data-types
amount | monetary amount, comprised of a numerical value and a currency
measure | numerical value associated with a measurement unit

So your application has to create those fields if an amount or a measure is found.
We do have a ticket to discuss this on our side to see if more could be done by the SDK itself, or to document this better, for example by implementing this in the eforms editor demo.
